sqlglot/dialects/hive.py

```diff
@@ -599,15 +599,52 @@ class Generator(generator.Generator):
 
         PROPERTIES_LOCATION = {
             **generator.Generator.PROPERTIES_LOCATION,
+            exp.DistributedByHash: exp.Properties.Location.POST_SCHEMA,
+            exp.DistributedByRandom: exp.Properties.Location.UNSUPPORTED,
+            exp.DuplicateKey: exp.Properties.Location.UNSUPPORTED,
+            exp.EngineProperty: exp.Properties.Location.UNSUPPORTED,
             exp.FileFormatProperty: exp.Properties.Location.POST_SCHEMA,
             exp.PartitionedByProperty: exp.Properties.Location.POST_SCHEMA,
+            exp.PrimaryKey: exp.Properties.Location.UNSUPPORTED,
             exp.VolatileProperty: exp.Properties.Location.UNSUPPORTED,
             exp.WithDataProperty: exp.Properties.Location.UNSUPPORTED,
         }
 
+        # https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL
+        def distributedbyhash_sql(self, expression: exp.DistributedByHash) -> str:
+            expressions = self.expressions(expression, key="expressions", flat=True)
+            sorted_by = self.expressions(expression, key="sorted_by", flat=True)
+            sorted_by = f" SORTED BY ({sorted_by})" if sorted_by else ""
+            buckets = "1"
+            if expression.auto_bucket:
+                self.unsupported(
+                    "DistributedByHash without buckets, Spark requires a number of buckets"
+                )
+            else:
+                buckets = self.sql(expression, "buckets")
+            return f"CLUSTERED BY ({expressions}){sorted_by} INTO {buckets} BUCKETS"
+
+        def doris_or_sr_type_to_hive(self, expression: exp.DataType) -> str:
```

Comment:
What is this?

Reply:
I want to convert the obtained dialect（include doris、mysql、starrocks） into a type supported by Hive, such as int(11) to int. As you said, I also think this is a hack. I will reconsider this part.